backend/src/utils/resourceUtils.ts

```diff
@@ -90,7 +90,7 @@ const fetchDashboardCR = async (fastify: KubeFastifyInstance): Promise<Dashboard
     )
     .then((res) => {
       const dashboardCR = res?.body as DashboardConfig;
-      return [dashboardCR];
+      return [_.merge({}, blankDashboardCR, dashboardCR)]; // merge with blank CR to prevent any missing values
```

Comment:
btw, I needed to edit this same file in another PR, it just bothers me that we are using both async/await and promises at the same time, it's kinda hard to read and it could lead to issues, I would say we either use one or another if you want I can rework my PR and you can piggyback into those changes.

Reply:
Sure. What pr are you talking about. I can switch my implementation